src/synaptik.js

```diff
@@ -177,3 +176,48 @@ export function connect(selector, lifecycle = {}) {
     );
   };
 }
+
+export const useSynaptik = selector => {
+  const vault = useContext(Context);
+  const [state, setState] = useState(selector(vault.stores));
+
+  useEffect(() => {
+    return vault.subscribe(() => {
+      const nextState = selector(vault.stores);
+      if (!isShallowEqual(nextState, state)) {
```

Comment:
Will we ever want to check for deep equality?

Reply:
I don't think so. I'd like to stick to the functionality of the current `PureComponent` implementation which only does the shallow reference and equality check